montecarlo/mcpy/plotting.py

```diff
@@ -0,0 +1,244 @@
+            lb = np.array([param_estimates[dgp_name][method_name][i][1][0] for i in range(len(param_estimates[dgp_name][method_name]))])
+            ub = np.array([param_estimates[dgp_name][method_name][i][1][1] for i in range(len(param_estimates[dgp_name][method_name]))])
+            lb_ = np.min(lb, axis=0)
+            ub_ = np.max(ub, axis=0)
+            plt.fill_between(X_test.reshape(100,), lb_, ub_, alpha=0.25)
+
+        true = true_params[dgp_name][0]
+        plt.plot(X_test, true, label='true effect')
+        plt.legend()
+        plt.show()
+
+def plot_violin(plot_name, param_estimates, metric_results, true_params, config):
+    for dgp_name, dgp_fn in metric_results.items():
+        n_methods = len(list(dgp_fn.keys()))
+        for metric_name in next(iter(dgp_fn.values())).keys():
+            if metric_name in config['per_plots']:
+                plt.figure(figsize=(1.5 * n_methods, 2.5))
+                plt.violinplot([dgp_fn[method_name][metric_name] for method_name in dgp_fn.keys()], showmedians=True)
+                plt.xticks(np.arange(1, n_methods + 1), list(dgp_fn.keys()))
+                plt.ylabel(metric_name)
+                plt.tight_layout()
+                plt.show()
+
+def plot_subset_param_histograms(param_estimates, metric_results, config, subset):
+    for dgp_name, pdgp in param_estimates.items():
+        n_methods = len(list(pdgp.keys()))
+        n_params = config['dgp_opts']['kappa_gamma'] + 1
+        plt.figure(figsize=(4 * n_params, 2 * n_methods))
+        for it, m_name in enumerate(pdgp.keys()):
+            for inner_it, i in enumerate(subset):
+                plt.subplot(n_methods, n_params, it * n_params + inner_it + 1)
+                plt.hist(pdgp[m_name][:, i])
+                plt.title("{}[{}]. $\\mu$: {:.2f}, $\\sigma$: {:.2f}".format(m_name, i, np.mean(pdgp[m_name][:, i]), np.std(pdgp[m_name][:, i])))
+        plt.tight_layout()
+        plt.savefig(os.path.join(config['target_dir'], 'dist_dgp_{}_{}.png'.format(dgp_name, config['param_str'])), dpi=300)
+        plt.close()
+    return
+
+def plot_param_histograms2(param_estimates, metric_results, config):
+    for dgp_name, pdgp in param_estimates.items():
+        n_methods = len(list(pdgp.keys()))
+        n_params = next(iter(pdgp.values())).shape[1]
+        plt.figure(figsize=(4 * n_params, 2 * n_methods))
+        for it, m_name in enumerate(pdgp.keys()):
+            for i in range(pdgp[m_name].shape[1]):
+                plt.subplot(n_methods, n_params, it * n_params + i + 1)
+                plt.hist(pdgp[m_name][:, i])
+                plt.title("{}[{}]. $\\mu$: {:.2f}, $\\sigma$: {:.2f}".format(m_name, i, np.mean(pdgp[m_name][:, i]), np.std(pdgp[m_name][:, i])))
+        plt.tight_layout()
+        plt.savefig(os.path.join(config['target_dir'], 'dist_dgp_{}_{}.png'.format(dgp_name, config['param_str'])), dpi=300)
+        plt.close()
+    return
+
+def plot_metrics2(param_estimates, metric_results, config):
```

Comment:
This isn't an intuitive name to me.

Also, add docstrings to all functions here.

Reply:
Many of these are from previous monte carlo code that isn't exactly extended to here,because the dgps and methods we're working with require the return types of the results to be specific to the dgp/method/metric. Thus many of these plotting functions are somewhat "deprecated". However, if someone's dgp/method/metrics fit a more basic form, these plotting functions can be used. I think I'll take them out for now and they can be added back later. It doesn't make sense to have them in here unless we have example dgps/methods/metrics that use them.  In general there's some more thinking that needs to be done for how to keep the framework general but still allow for metrics and plots to specific forms of dgp/method. The best way to do that is to just implement dgps/methods/metrics until there's a clash and one realizes the framework needs to be slightly adjusted.